src/docker-compose.ts

```diff
@@ -0,0 +1,427 @@
+   * Service descriptions.
+   */
+  readonly services?: Record<string, DockerComposeServiceDescription>;
+}
+
+/**
+ * Create a docker-compose YAML file.
+ */
+export class DockerCompose extends Component {
+  /**
+   * Create a port mapping
+   * @param publishedPort Published port number
+   * @param targetPort Container's port number
+   * @param protocol Port's protocol (Default: TCP)
+   */
+  static portMapping(
+    publishedPort: number,
+    targetPort: number,
+    protocol: DockerComposeProtocol = DockerComposeProtocol.TCP,
+  ): DockerComposeServicePort {
+    return {
+      target: targetPort,
+      published: publishedPort,
+      protocol: protocol,
+      mode: 'host',
+    };
+  }
+
+  /**
+   * Create a bind volume that binds a host path to the target path in the container.
+   * @param sourcePath Host path name
+   * @param targetPath Target path name
+   */
+  static bindVolume(sourcePath: string, targetPath: string): IDockerComposeVolumeBinding {
+    return {
+      bind(_volumeInfo: IDockerComposeVolumeConfig): DockerComposeVolumeMount {
+        return {
+          type: 'bind',
+          source: sourcePath,
+          target: targetPath,
+        };
+      },
+    };
+  }
+
+  /**
+   * Create a named volume and mount it to the target path. If you use this
+   * named volume in several services, the volume will be shared. In this
+   * case, the volume configuration of the first-provided options are used.
+   *
+   * @param volumeName Name of the volume
+   * @param targetPath Target path
+   * @param options volume configuration (default: docker compose defaults)
+   */
+  static namedVolume(volumeName: string, targetPath: string, options: DockerComposeVolumeConfig = {}): IDockerComposeVolumeBinding {
+    return {
+      bind(volumeInfo: IDockerComposeVolumeConfig): DockerComposeVolumeMount {
+        volumeInfo.addVolumeConfiguration(volumeName, options);
+
+        return {
+          type: 'volume',
+          source: volumeName,
+          target: targetPath,
+        };
+      },
+    };
+  }
+
+  private readonly services: Record<string, DockerComposeServiceDescription>;
+
+  constructor(project: Project, props?: DockerComposeProps) {
+    super(project);
+
+    const nameSuffix = props?.nameSuffix ? `${props!.nameSuffix}.yml` : 'yml';
+    new YamlFile(project, `docker-compose.${nameSuffix}`, {
+      committed: true,
+      readonly: true,
+      obj: () => this._synthesizeDockerCompose(),
+    });
+
+    this.services = {};
+
+    // Add the services provided via the constructor argument.
+    const initialServices = props?.services ?? {};
+    for (const [name, serviceDescription] of Object.entries(initialServices)) {
+      this.addService(name, serviceDescription);
+    }
+  }
+
+  /**
+   * Add a service to the docker-compose file.
+   * @param name name of the service
+   * @param description a service description
+   */
+  addService(name: string, description: DockerComposeServiceDescription): void {
```

Comment:
Specify access modifier:

```suggestion
  public addService(name: string, description: DockerComposeServiceDescription): void {
```